modules/TransformationPlugin/src/main/java/org/gephi/transformation/plugin/operations/Mirror.java

```diff
@@ -0,0 +1,30 @@
+package org.gephi.transformation.plugin.operations;
+
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.Node;
+import org.gephi.transformation.spi.TransformationOperation;
+
+
+abstract public class Mirror implements TransformationOperation {
+    protected Function<Node, Float> nodeGetFunction;
+    protected BiConsumer<Node, Float> nodeSetFunction;
+
+
+    @Override
+    public void transformation(Graph graph) {
+        float barycenter = 0.f;
+        for (Node n : graph.getNodes()) {
+            barycenter += nodeGetFunction.apply(n);
+        }
+        barycenter /= graph.getNodeCount();
+        for (Node node : graph.getNodes()) {
+            if (node.isFixed()) {
+                return;
```

Comment:
Looks weird to return here. On top of it, this would NOT release the (auto) read lock, which is a major issue